wallet/src/main/scala/org/bitcoins/wallet/models/AddressRecord.scala

```diff
@@ -0,0 +1,135 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.hd.{
+  HDChainType,
+  HDCoinType,
+  HDPurpose,
+  HDPurposes,
+  LegacyHDPath,
+  NestedSegWitHDPath,
+  SegWitHDPath
+}
+import org.bitcoins.core.protocol.{
+  Bech32Address,
+  BitcoinAddress,
+  P2PKHAddress,
+  P2SHAddress
+}
+import org.bitcoins.core.protocol.script.{ScriptPubKey, ScriptWitness}
+import org.bitcoins.crypto.{ECPublicKey, Sha256Hash160Digest}
+
+case class AddressRecord(
+    purpose: HDPurpose,
+    accountCoin: HDCoinType,
+    accountIndex: Int,
+    accountChain: HDChainType,
+    addressIndex: Int,
+    address: BitcoinAddress,
+    pubKey: ECPublicKey,
+    hashedPubKey: Sha256Hash160Digest,
+    scriptPubKeyId: Long,
+    scriptWitnessOpt: Option[ScriptWitness]
```

Comment:
Do we also want a redeem script option for P2SH?

Reply:
Potentially. But currently I'd like to make it working at least this way.

Reply:
We shouldn't need the redeem script or script witness for the address. It will fill in these accordingly for the spending info db.
https://github.com/bitcoin-s/bitcoin-s/blob/master/wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala#L200